app/pages/Account/functions.js

```diff
@@ -0,0 +1,256 @@
+    if (revealCoin) {
+      // Is local?
+      if (revealCoin.height < auction.height) {
+        continue;
+      }
+      redeemableHNS += bid.value;
+      redeemableNum++;
+    }
+  }
+
+  // locked Bidding = bids in BIDDING state auctions
+  // locked Reveal = bids in REVEAL state auctions (includes blind if not revealed yet)
+  // locked Finished = bids that can be REDEEMed or REGISTERed
+  return {
+    lockedBalance: {
+      bidding: { HNS: biddingHNS, num: biddingNum },
+      revealing: { HNS: revealingHNS, num: revealingNum },
+      finished: {
+        HNS: registerableHNS + redeemableHNS,
+        num: registerableNum + redeemableNum,
+      },
+    },
+    actionableInfoBids: {
+      revealable: {
+        HNS: revealableHNS,
+        num: revealableNum,
+        block: revealableBlock,
+      },
+      redeemable: { HNS: redeemableHNS, num: redeemableNum },
+      registerable: { HNS: registerableHNS, num: registerableNum },
+    },
+  };
+}
+
+async function getStatsFromNames() {
+  let transferringDomains = new Set();
+  let transferringBlock = null;
+  let finalizableDomains = new Set();
+  let renewableDomains = new Set();
+  let renewableBlock = null;
+
+  const names = await walletClient.getNames();
+  console.log("names", names);
+
+  for (let name of names) {
+    const auction = await walletClient.getAuctionInfo(name.name);
+    console.log(auction);
+
+    const ownerCoin = await walletClient.getCoin(
+      auction.owner.hash,
+      auction.owner.index
+    );
+    if (!ownerCoin) {
+      continue;
+    }
+
+    // Mark for renew if the name is going to expire in the next 2 months
+    if (auction.stats.daysUntilExpire < 30 * 2) {
+      renewableDomains.add(auction.name);
+      if (
+        renewableBlock === null ||
+        auction.stats.blocksUntilExpire < renewableBlock
+      ) {
+        renewableBlock = auction.stats.blocksUntilExpire;
+      }
+    }
+
+    // Names being transferred?
+    if (auction.transfer !== 0) {
+      // Either finalizable now, or not
+      if (auction.stats.blocksUntilValidFinalize <= 0) {
+        finalizableDomains.add(auction.name);
+      } else {
+        transferringDomains.add(auction.name);
+        if (
+          transferringBlock === null ||
+          auction.stats.blocksUntilValidFinalize < transferringBlock
+        ) {
+          transferringBlock = auction.stats.blocksUntilValidFinalize;
+        }
+      }
+    }
+  }
+
+  return {
+    actionableInfoNames: {
+      renewable: {
+        domains: renewableDomains,
+        block: renewableBlock,
+      },
+      transferring: {
+        domains: transferringDomains,
+        block: transferringBlock,
+      },
+      finalizable: { domains: finalizableDomains },
+    },
+  };
+}
+
+async function getUsdConversion() {
+  try {
+    const response = await (
+      await fetch(
+        "https://api.coingecko.com/api/v3/simple/price?ids=handshake&vs_currencies=usd"
```

Comment:
could this be a setting? usd/cad/eur etc

Reply:
Hm yeah, can do that :+1: